_posts/2022-01-03-contributing-for-the-first-time-to-a-project.adoc

```diff
@@ -0,0 +1,174 @@
+Since the very beginning developers need to properly set up the local environment including multiple:
+
+- build tools
+- runtimes
+- dependencies
+- IDEs
+
+For complex projects, it could easily take weeks for new developers to get the builds running on their machines.
+In general, a sufficient amount of the team's week is spent on building and maintaining developer environments.
+With {prod} we are solving the complexity of the onboarding, configuration, and setup by defining a new cloud-native, zero-install, and repeatable concept of a workspace.
+
+=== Contributing for the first time to a project using {prod}
+
+The motto of {prod} is that anyone anytime can contribute to a project without installing software.
+In order to contribute to a project, just click on the badge defined in the `README.md` file of the repository, and within a minute you will get a workspace ready for development with all the tools and runtimes available out of the box.
+
+.The {prod} contribute badge in `README.md`
+image::/assets/img/contribute-to-a-project/contribute.png[Contribute Badge]
+{empty}
+
+By default {prod} is using a customized version of link:https://theia-ide.org/[Theia], which provides link:https://code.visualstudio.com/[VS Code] like experience including the support of extensions:
+
+.{prod} Theia IDE
+image::/assets/img/contribute-to-a-project/ide.png[{prod} Theia IDE]
+{empty}
+
+Programming languages facilities like code completion, navigation, and codelens are provided using the link:https://microsoft.github.io/language-server-protocol/[Language Server Protocol]. {prod} benefits from a growing ecosystem of over 60 available language servers, delivering intelligent editing support for all major programming languages. 
+
+.Java Content Assist
+image::/assets/img/contribute-to-a-project/content-assist.png[Content Assist]
+{empty}
+
+Pre-defined commands (`Terminal` -> `Run Task...`) allow you to easily build, run, debug, and test the application.
+
+.Pre-defined commands
+image::/assets/img/contribute-to-a-project/commands.png[IDE Commands]
+{empty}
+
+Additional extensions can be enabled using `Ctrl+Shift+J` shortcut or `View` -> `Plugins` menu items:
+
+.The 'Plugins' view
+image::/assets/img/contribute-to-a-project/plugins.png[The 'Plugins' view]
+{empty}
+
+Debug the application using the `F5` shortcut or `Run` -> `Start Debugging` menu items:
+
+.Debugging Java application
+image::/assets/img/contribute-to-a-project/debug.png[Debugging the application]
+{empty}
+
+Commit and push the changes from the terminal or use the `Source Control` view:
+
+.The 'Source Control' view
+image::/assets/img/contribute-to-a-project/source-control.png[[The 'Source Control' view]
+{empty}
+
+Now you are ready for the first contribution. For GitHub, the dedicated plugin is available that allows opening pull requests right from the IDE:
+
+.The 'GitHub' plugin
+image::/assets/img/contribute-to-a-project/github-plugin.png[[The 'GitHub' plugin]
+{empty}
+
```

Comment:
The plugin currently does not work in the DevWorkspace mode https://github.com/eclipse/che/issues/20860